src/PlantBundle/Controller/PlantController.php

```diff
@@ -30,7 +42,51 @@ public function showAction($id)
 
         $plant= $em->getRepository('PlantBundle:Plant')->find($id);
 
+        if ($plant==null)
+        {
+            $this->addFlash('error', 'Could not find plant');
+
+            return $this->redirectToRoute('homepage');
+        }
         return $this->render('plant/show.html.twig', ['plant' => $plant]);
     }
+
+    /**
+     * @Route("/plant/add", name="addPlant")
+     */
+    public function addAction(Request $request)
+    {
+        if ($this->get('security.authorization_checker')->isGranted('ROLE_USER'))
```

Comment:
This approach is better, but not the best. You should check if user is **not** logged in. If unlogged, then save flash message and redirect. Then You do not need `else` part.
It's way better because You do not have to wrap code in giant `if`-`else` condition then. Just small `if`.